plugins/extpoints/interfaces.go

```diff
@@ -0,0 +1,41 @@
+package extpoints
+
+import (
+	"sync"
+
+	"github.com/taskcluster/taskcluster-worker/engines"
+	p "github.com/taskcluster/taskcluster-worker/plugins"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+// The Plugin Factory is responsible for maintaining a list of plugin factories
+// that are used to create new instances of a Plugin for a task execution.
+type PluginFactory func(options *p.PluginOptions) p.Plugin
```

Comment:
Nope... This is bad... It's also why I renamed `PluginFactory` to `PluginEnvironment` (which still isn't a good name)...

If this is a cache folder plugin, then `PluginFactory` will hold a list of all idle cache folders... If there is just one `PluginFactory` for the cache folder plugin in the entire process then it uses global variables... I'll never need two, but it's nice to have the option for testing...

Ie. `PluginFactory` is stateful therefore we should have the option of having more than one, for the same kind of plugin... Only useful in testing, but still useful.